src/zjs_modules.c

```diff
@@ -254,6 +264,53 @@ static ZJS_DECL_FUNC(process_exit)
     exit(status);
 }
 #endif
+#ifdef ZJS_DYNAMIC_LOAD
+void zjs_modules_check_load_file(char *file)
+{
+    // No file waiting to load, just return
+    if (load_file == NULL) {
+        return;
+    }
+
+    zjs_stop_js();
+    char *buf = NULL;
+    size_t size;
+    jerry_value_t parsed_code = 0;
+    buf = read_file_alloc(load_file, &size);
+    parsed_code = jerry_parse((const jerry_char_t *)buf, size, false);
+
+    if (jerry_value_has_error_flag(parsed_code)) {
+     ERR_PRINT("Error parsing JS\n");
+    }
+
+    zjs_free(buf);
+    ZVAL ret_value = jerry_run(parsed_code);
+    if (jerry_value_has_error_flag(ret_value)) {
+     ERR_PRINT("Error running JS\n");
+    }
+
+    // Remove the load file so it doesn't load it again
+    zjs_free(load_file);
+    load_file = NULL;
+    jerry_release_value(parsed_code);
+}
+
+static ZJS_DECL_FUNC(zjs_run_js)
+{
+    ZJS_VALIDATE_ARGS(Z_STRING);
+    size_t len = 15;
+    size_t file_len;
+    load_file = zjs_malloc(len);
+
+    zjs_copy_jstring(argv[0], load_file, &file_len);
```

Comment:
Why 15? On A101 they're limited to 8.3 I think, you would need 13 bytes including null terminator? But maybe better to not enforce that limit here and just use `zjs_alloc_from_jstring` which is simpler. Then `fs_exist` could change later (or platform-dependently) to allow different sized filenames. You do need to check for NULL on `load_file` here either way though.

Reply:
I was assuming this code would be used on devices other than A101.  15 was arbitrary, I'll either use a globally defined max str length or create one.